spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -829,6 +831,108 @@ When a repository method is called with a null value as an argument to one of it
 
 NOTE: The behavior of a query method when the method is called with a null argument is not defined by this specification, and is not portable between Jakarta Data providers.
 
+==== Restriction parameters
+
+The `Restriction<T>` type is a special parameter used to dynamically construct queries in a fluent and type-safe manner. Valid as parameters to methods that perform queries, such as methods annotated `@Find` or `@Delete`, restrictions allow the application to define additional conditions programmatically, often using a generated metamodel class (see <<metamodel-java-processor>>).
+
+When using `Restriction<T>` in repository methods:
+
+- The method *must* be annotated with either `@Find` or `@Query`. Without one of these annotations, the use of `Restriction<T>` is *not valid* and results in a `MappingException`.
+- When `@Find` or `@Query` is present, `Restriction<T>` can appear in *any position* in the parameter list. In these cases, the `Restriction<T>` provides the full query logic programmatically, and any query derivation from the method name is ignored.
+
+For example:
+
+[source,java]
+----
+@Query("WHERE (product.numAvailable > 0)")
+List<Product> search(Restriction<Product> restriction, Order<Product> order);
+
+@Find
+Page<Product> findAvailable(Restriction<Product> restriction, PageRequest pageRequest);
+----
+
+The following is *invalid* and will throw a `MappingException`:
+
+[source,java]
+----
+List<Product> findByName(Restriction<Product> restriction);
+----
+
+In both cases, Restriction<T> must not be used with lifecycle annotations such as @Save, @Insert, @Update, or @Delete.
```

Comment:
To avoid confusion over Delete, which is sometimes a lifecycle method and sometimes performs a delete query depending on the parameter.
```suggestion
In both cases, Restriction<T> must not be used with lifecycle methods.
```